app/controllers/dashboard/form/publish_controller.rb

```diff
@@ -24,12 +24,19 @@ def update
         # against the draft validations, then mark the record as published and
         # save again--this time using the published validations. That way the
         # appropriate error messages will appear on the form when it's re-rendered
-        if publish?
+        if publish? && !@resource.draft_curation_requested
           # WorkVersion#set_thumbnail_selection may be unreliable if the Shrine::ThumbnailJob is delayed
           @resource.set_thumbnail_selection
           @resource.indexing_source = Proc.new { nil }
           @resource.save
           @resource.publish
+        elsif request_curation?
+          begin
+            AirtableExporter.call(@resource.id)
+            @resource.draft_curation_requested = true
+          rescue Airrecord::Error
+            flash[:error] = t('dashboard.form.publish.curation.error')
```

Comment:
Just tested this out, and it works the same way as the save & exit button so users wouldn't see the validation error & Airrecord error at the same time. Validation error messages are contained entirely within the form, and users can save & exit/request curation even if they have an invalid entry. I assume this behavior is to allow data to persist so users can come back later & see where they went wrong. The Airrecord validation error displays after the redirect back to the work page. So it is possible a user could get a message that their work saved successfully as well as a message that the curation request failed. In this case, they are prompted to request curation again or contact us if the problem persists. I was debating how to handle this scenario & am open to suggestions.

Reply:
If I input a junk version_name and submit it shows both the Airtable error and validation error.  Would it make sense to only proceed with the Airtable export if the record is valid + saves?

Reply:
Good question, @ajkiessl. I'm not sure what the right answer is. Thinking through it - the work/version records would have already been created at this point with at least some of the metadata entered. A validation error at this step would indicate that something is missing or incorrect that would prevent the work from being published. My assumption is that we'd want the depositor to give their best shot at submitting complete/publishable files and metadata _before_ the work is sent off for curation. With that being the case, I think we'd want to do what you suggested and only proceed with the Airtable export if the record is valid and save successfully. However, it would need to be clear to the depositor that their curation request didn't succeed due to the validation errors and that they would need to retry after fixing the errors.